.github/workflows/e2e.yml

```diff
@@ -0,0 +1,61 @@
+name: E2E
+
+on:
+  pull_request:
+    branches: [ master, release/** ]
```

Comment:
I noticed these E2E tests take 52 minutes to complete. Is that really something we want to run for every PR commit or should it be a nightly job or something else? @rbair23 @tinker-michaelj WDYT?

Reply:
There are definitely trade-offs, but I don't think we can currently go without this sanity-check on a PR to `master`.